ihmc-high-level-behaviors/src/libgdx/java/us/ihmc/rdx/ui/teleoperation/RDXRobotLowLevelMessenger.java

```diff
@@ -1,38 +1,101 @@
 package us.ihmc.rdx.ui.teleoperation;
 
+import controller_msgs.msg.dds.EnableHPUCommandMessage;
+import controller_msgs.msg.dds.EnableHPUStatusMessage;
 import controller_msgs.msg.dds.GoHomeMessage;
+import controller_msgs.msg.dds.MasterGainScaleControllerCommandMessage;
+import controller_msgs.msg.dds.MasterGainScaleControllerStatusMessage;
 import controller_msgs.msg.dds.StopAllTrajectoryMessage;
 import imgui.ImGui;
-import imgui.type.ImInt;
 import us.ihmc.behaviors.tools.CommunicationHelper;
 import us.ihmc.communication.controllerAPI.RobotLowLevelMessenger;
+import us.ihmc.rdx.imgui.ImGuiTools;
 import us.ihmc.rdx.imgui.ImGuiUniqueLabelMap;
 import us.ihmc.rdx.ui.RDXBaseUI;
+import us.ihmc.tools.Timer;
 
 public class RDXRobotLowLevelMessenger
 {
    private final ImGuiUniqueLabelMap labels = new ImGuiUniqueLabelMap(getClass());
    private final RobotLowLevelMessenger robotLowLevelMessenger;
    private final CommunicationHelper communicationHelper;
-   private final RDXTeleoperationParameters teleoperationParameters;
-   private final ImInt pumpPSI = new ImInt(1);
-   private final String[] psiValues = new String[] {"1500", "2300", "2500", "2800"};
+   private final Timer hpuConnectedTimer = new Timer();
+   private final Timer robotServoedConnectedTimer = new Timer();
+   private boolean hpuEnabled = false;
+   private boolean isRobotServoed = false;
 
-   public RDXRobotLowLevelMessenger(CommunicationHelper communicationHelper, RDXTeleoperationParameters teleoperationParameters)
+   public RDXRobotLowLevelMessenger(CommunicationHelper communicationHelper)
    {
       this.communicationHelper = communicationHelper;
-      this.teleoperationParameters = teleoperationParameters;
-      robotLowLevelMessenger = communicationHelper.getOrCreateRobotLowLevelMessenger();
 
+      robotLowLevelMessenger = communicationHelper.getOrCreateRobotLowLevelMessenger();
       if (robotLowLevelMessenger == null)
       {
          String robotName = communicationHelper.getRobotModel().getSimpleRobotName();
          throw new RuntimeException("Please add implementation of RobotLowLevelMessenger for " + robotName);
       }
+
+      communicationHelper.subscribeToControllerViaVolatileCallback(EnableHPUStatusMessage.class, message ->
+      {
+         hpuConnectedTimer.reset();
+         hpuEnabled = message.getHpuEnabled();
+      });
+      communicationHelper.subscribeToControllerViaVolatileCallback(MasterGainScaleControllerStatusMessage.class, message ->
+      {
+         robotServoedConnectedTimer.reset();
+         isRobotServoed = message.getIsRobotServoed();
+      });
    }
 
    public void renderImGuiWidgets()
    {
```

Comment:
My idea here was to only provide the safe buttons. So only "Enable HPU", "Servo robot", and "Unservo slowly".